src/jupyterhub_config/custom_docker_spawner.py

```diff
@@ -229,24 +231,42 @@ def _configure_environment(self, user_dir: Path, user_env_dir: Path, username: s
 
         self.environment["SHELL"] = "/usr/bin/bash"
 
-        if self._is_rw_minio_user():
-            self.log.info(
-                f"MinIO read/write user detected: {self.user.name}. Setting up minio_rw credentials."
+        # Get user-specific MinIO credentials from governance service
+        # Check if KBASE_AUTH_TOKEN is available (should be set by pre_spawn_start)
+        kbase_token = self.environment.get("KBASE_AUTH_TOKEN")
+        if not kbase_token:
+            raise ValueError(
+                "KBASE_AUTH_TOKEN not found in spawner environment. Authentication required for governance client."
             )
-            self.environment["MINIO_ACCESS_KEY"] = self.environment[
-                "MINIO_RW_ACCESS_KEY"
-            ]
-            self.environment["MINIO_SECRET_KEY"] = self.environment[
-                "MINIO_RW_SECRET_KEY"
-            ]
-            # USAGE_MODE is used by the setup.sh script to determine the appropriate configuration for the user.
-            self.environment["USAGE_MODE"] = "dev"
+        client = DataGovernanceClient(kbase_token=kbase_token)
+
+        # Get credentials for the current user - it will also create a new user if it doesn't exist
+        credentials = client.get_credentials()
+        self.log.info(
+            f"Retrieved governance credentials for user {username}: access_key='{credentials.access_key}'"
+        )
+
+        if self.environment.get("USE_DATA_GOVERNANCE_CREDENTIALS", "false") == "true":
+            # Set user-specific MinIO credentials
+            self.environment["MINIO_ACCESS_KEY"] = credentials.access_key
+            self.environment["MINIO_SECRET_KEY"] = credentials.secret_key
```

Comment:
I assume the user will need to get these creds to log in to the minio UI. How do they do that?

Reply:
currently they do `os.environ['MINIO_ACCESS_KEY']` accordingly to the [guide](https://github.com/kbase/cdm-jupyterhub/blob/main/docs/dev_guide.md#readwrite-minio-username-and-password). I'm planning to implement a magic command to make it more convenient for users.